flutter/lib/src/integrations/native_app_start_handler.dart

```diff
@@ -0,0 +1,303 @@
+    transaction.children.addAll([
+      appStartSpan,
+      pluginRegistrationSpan,
+      sentrySetupSpan,
+      firstFrameRenderSpan
+    ]);
+  }
+
+  Future<void> _attachNativeSpans(
+    AppStartInfo appStartInfo,
+    SentryTracer transaction,
+    SentrySpan parent,
+  ) async {
+    await Future.forEach<TimeSpan>(appStartInfo.nativeSpanTimes,
+        (timeSpan) async {
+      try {
+        final span = await _createAndFinishSpan(
+          tracer: transaction,
+          operation: appStartInfo.appStartTypeOperation,
+          description: timeSpan.description,
+          parentSpanId: parent.context.spanId,
+          traceId: transaction.context.traceId,
+          startTimestamp: timeSpan.start,
+          endTimestamp: timeSpan.end,
+        );
+        span.data.putIfAbsent('native', () => true);
+        transaction.children.add(span);
+      } catch (e) {
+        _options.logger(SentryLevel.warning,
+            'Failed to attach native span to app start transaction: $e');
+      }
+    });
+  }
+
+  Future<SentrySpan> _createAndFinishSpan({
+    required SentryTracer tracer,
+    required String operation,
+    required String description,
+    required SpanId parentSpanId,
+    required SentryId traceId,
+    required DateTime startTimestamp,
+    required DateTime endTimestamp,
+  }) async {
+    final span = SentrySpan(
+      tracer,
+      SentrySpanContext(
+        operation: operation,
+        description: description,
+        parentSpanId: parentSpanId,
+        traceId: traceId,
+      ),
+      _hub,
+      startTimestamp: startTimestamp,
+    );
+    await span.finish(endTimestamp: endTimestamp);
+    return span;
+  }
+}
+
+enum AppStartType { cold, warm }
+
+class AppStartInfo {
+  AppStartInfo(
+    this.type, {
+    required this.start,
+    required this.pluginRegistration,
+    required this.sentrySetupStart,
+    required this.nativeSpanTimes,
+    this.end,
+  });
+
+  final AppStartType type;
+  final DateTime start;
+  final List<TimeSpan> nativeSpanTimes;
+
+  // We allow the end to be null, since it might be set at a later time
+  // with setAppStartEnd when autoAppStart is disabled
+  DateTime? end;
+
+  final DateTime pluginRegistration;
+  final DateTime sentrySetupStart;
+
+  Duration? get duration => end?.difference(start);
+
+  SentryMeasurement? toMeasurement() {
+    final duration = this.duration;
+    if (duration == null) {
+      return null;
+    }
+    return type == AppStartType.cold
+        ? SentryMeasurement.coldAppStart(duration)
+        : SentryMeasurement.warmAppStart(duration);
+  }
+
+  String get appStartTypeOperation => 'app.start.${type.name}';
+
+  String get appStartTypeDescription =>
+      type == AppStartType.cold ? 'Cold Start' : 'Warm Start';
+  final pluginRegistrationDescription = 'App start to plugin registration';
+  final sentrySetupDescription = 'Before Sentry Init Setup';
+  final firstFrameRenderDescription = 'First frame render';
+}
+
+class TimeSpan {
```

Comment:
```suggestion
class _TimeSpan {
```

can be private if I'm not wrong